### PR TITLE
Paw/ct 1844 log params take 2

### LINKED ISSUE #6639: [CT-1844] Improve Granularity of Log Configuration
### Describe the feature

This is a proposal which would subsume the following four issues:
- https://github.com/dbt-labs/dbt-core/issues/6580
- https://github.com/dbt-labs/dbt-core/issues/6312
- https://github.com/dbt-labs/dbt-core/issues/5208
- https://github.com/dbt-labs/dbt-core/issues/6615

The proposal is to modify log configuration so that the following command line help would be accurate, allowing for independent file and console log configuration of color, format, and level:

```
  --log-format {text,json,default}
  --log-format-file {text,json,default}
                        Specify the log format, overriding the command's
                        default. The standard option will apply the format
                        to both console and file output, unless the -file
                        variant (which can also appear alone) is used to 
                        override the log file format.
   --log-level {debug, info, warn, error, none}
   --log-level-file {debug, info, warn, error, none}
                        Set the logging level. Dbt will log events at or above
                        the severity level specified. The standard option will
                        apply to both console and file output, unless the
                        -file variant (which can also appear alone) is used to 
                        override the log file format. The 'none' level disables
                        all log output, including log file creation.
  --use-colors          Colorize (or do not colorize when using the 'no' 
  --use-colors-file     variants) dbt's log output. Output is colorized by
  --no-use-colors       by default and may also be set in a profile or at 
  --no-use-colors-file  the command line. The standard option will apply to
                        to both console and file output, unless the -file
                        variant (which can also appear alone) is used to 
                        override the log file colorization.
```

### PR DESCRIPTION
**Pull request by peterallenwebb**
resolves #6639

### Description

Adds the following cli parameters for finer grained control of console and file log behavior, as described in #6639:

--log-format
--log-format-file
--log-level
 --log-level-file
--use-colors 
--use-colors-file
--no-use-colors
--no-use-colors-file  the 

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by gshank** [comment]
This doesn't touch UserConfig (in core/dbt/contracts/project.py) where users can set log_format and use_colors. So I'm wondering how those settings integrate with these cli params, and whether we shouldn't allow setting the new params (somehow) in user config also. If I have a logging configuration that I want to use all the time, I would prefer not to have to pass in cli params every time, but have the settings persistent in user config.

**Comment by peterallenwebb** [comment]
@gshank Yes, that is an interesting point which I did not consider. 

@jtcohen6 Do you agree that we should let users set these same flags in their user config, following the pattern of log_format and use_colors? I worry that the interaction patterns between all the different places these get set could start to become complicated, but Gerda's observations do make sense to me.

On the other hand, maybe Gerda's use case could be supported by the work we are considering for #6993.

**Comment by jtcohen6** [comment]
@peterallenwebb @gshank Two-part answer:
1. Every "global config" that is settable via CLI flag + env var should also be settable in `UserConfig`. I know that is not the case today, but it is the documented behavior, and we ought to make it so. Let's do this for these logging parameters as well.
2. We should move `UserConfig` out of `profiles.yml`, and into its own standalone config file, in order to simplify the initialization flow for dbt-core. I agree it's a tangled web we weave today. My latest proposal for this: https://github.com/dbt-labs/dbt-core/issues/6207#issuecomment-1433007301

**Comment by peterallenwebb** [comment]
@gshank Still waiting for the checks to pass here, but let me know whether the latest changes address your concerns.

File: core/dbt/cli/flags.py [hunk 1]
```diff
@@ -172,6 +177,12 @@ def assign_params(ctx, params_assigned_from_default):
         for param in params:
             object.__setattr__(self, param.lower(), getattr(self, param))
 
+    # If the value of the lead parameter was set explicitly, apply the value to follow,
+    # unless follow was also set explicitly.
```
Comment (MichelleArk) [inline core/dbt/cli/flags.py:181]: tiny nit: documenting the method in a docstring below the signature would be more pythonic + consistent with the rest of this file.

File: core/dbt/events/functions.py [hunk 2]
```diff
@@ -18,103 +18,146 @@
...
 
-def setup_event_logger(
-    log_path: str,
-    log_format: str,
-    use_colors: bool,
-    debug: bool,
-    log_cache_events: bool,
-    quiet: bool,
-) -> None:
+def setup_event_logger(flags) -> None:
```
Comment (MichelleArk) [inline core/dbt/events/functions.py:21]: are we able to annotate the type of `flags` here?

File: core/dbt/events/functions.py [hunk 3]
```diff
@@ -18,103 +18,146 @@
...
-    else:
         EVENT_MANAGER.add_logger(
-            _get_stdout_config(log_format, debug, use_colors, log_cache_events, quiet)
+            _get_logbook_log_config(
+                flags.DEBUG, flags.USE_COLORS, flags.LOG_CACHE_EVENTS, flags.QUIET
+            )
         )
+    else:
+        if flags.LOG_LEVEL != "none":
+            log_format = _line_format_from_str(flags.LOG_FORMAT, LineFormat.PlainText)
```
Comment (MichelleArk) [inline core/dbt/events/functions.py:33]: is there a difference between `log_format` and `line_format`? If not, `line_format` would be more intuitive given the naming of `_line_format_from_str` + parameter name of `_get_stdout_config`.

**Review by MichelleArk — APPROVED** [review]
Just a couple small nits, otherwise LGTM!

File: core/dbt/events/functions.py (hunk 3 above)
Comment (peterallenwebb) [reply core/dbt/events/functions.py:33]: I agree, the variable being assigned here should be line_format. There is no difference between the two things, and the difference in naming is unfortunate.

**Pull request merged**